### PR TITLE
bump: transmission to use `neptune` only, drop `neptune-client`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

With the next feari=ure release let's cut and exclusively support `neptune` and drop legacy `neptune-client`, see: https://github.com/Lightning-AI/pytorch-lightning/pull/19131#discussion_r1428034943

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--19265.org.readthedocs.build/en/19265/

cc @borda

### CODE REVIEW

**Comment by awaelchli** [comment]
@Borda take a look a the response here from @SiddhantSadangi from the neptune team: https://github.com/Lightning-AI/pytorch-lightning/pull/18558#issuecomment-1721232187

**Comment by Borda** [comment]
> @Borda take a look a the response here from @SiddhantSadangi from the neptune team: [#18558 (comment)](https://github.com/Lightning-AI/pytorch-lightning/pull/18558#issuecomment-1721232187)

Yes, the response was half a year ago, but still, if they support it, the API switch is for legacy `Neptune-client`, so the API in the new versions is the same I beilve we can cut `Neptune-client<1.0`

**Comment by Borda** [comment]
so with installing `neptune==1.8.6` and asking import check
```
>>> import neptune
>>> from lightning_utilities.core.imports import RequirementCache
>>> RequirementCache("neptune>=1.0")
Requirement 'neptune>=1.0' met
```
then install the same `neptune-client==1.8.6` and doing the same
```
>>> import neptune
/mnt/69B27B700DDA7D73/venvs/lai/lib/python3.9/site-packages/neptune/internal/backends/hosted_client.py:51: NeptuneDeprecationWarning: The 'neptune-client' package has been deprecated and will be removed in the future. Install the 'neptune' package instead. For more, see https://docs.neptune.ai/setup/upgrading/
  from neptune.version import version as neptune_client_version
>>> from lightning_utilities.core.imports import RequirementCache
>>> RequirementCache("neptune>=1.0")
DistributionNotFound: The 'neptune>=1.0' distribution was not found and is required by the application. HINT: Try running `pip install -U 'neptune>=1.0'`
>>> from lightning_utilities.core.imports import module_available
>>> module_available("neptune")
True
```

File: src/lightning/pytorch/loggers/neptune.py [hunk 1]
```diff
@@ -223,8 +226,9 @@ def __init__(
         prefix: str = "training",
         **neptune_run_kwargs: Any,
     ):
-        if not _NEPTUNE_AVAILABLE and not _NEPTUNE_CLIENT_AVAILABLE:
-            raise ModuleNotFoundError(str(_NEPTUNE_AVAILABLE))
+        if not _NEPTUNE_AVAILABLE:
+            raise ModuleNotFoundError("Missing Neptune logger, please install it with `pip install neptune`")
```
Comment (awaelchli) [inline src/lightning/pytorch/loggers/neptune.py:230]: The previous `str(_NEPTUNE_AVAILABLE)` for printing was fine, no? We do it everywhere in Lightning

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: src/lightning/pytorch/loggers/neptune.py (hunk 1 above)
Comment (Borda) [reply src/lightning/pytorch/loggers/neptune.py:230]:
```suggestion
            raise ModuleNotFoundError(str(_NEPTUNE_AVAILABLE))
```

**Review by Raalsky — APPROVED** [review]

**Comment by Raalsky** [comment]
Thanks @Borda for handling this :wink:. LGTM 🚀 . We should probably mention this PR here as well: https://github.com/Lightning-AI/pytorch-lightning/issues/18555 as it's not going to be fixed according to these PR. We're going to send some communication to our clients about the discontinuation of this support.

I'll rebase all my PRs soon after merge of this one :wink:.

**Pull request merged**